subsys/sensing/sensor_mgmt.h

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SENSOR_MGMT_H_
+#define SENSOR_MGMT_H_
+
+#include <zephyr/sensing/sensing_datatypes.h>
+#include <zephyr/sensing/sensing_sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/slist.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define PHANDLE_DEVICE_BY_IDX(idx, node, prop)				\
+	DEVICE_DT_GET(DT_PHANDLE_BY_IDX(node, prop, idx))
+
+#define PHANDLE_DEVICE_LIST(node, prop)					\
+{									\
+	LISTIFY(DT_PROP_LEN_OR(node, prop, 0),				\
+			PHANDLE_DEVICE_BY_IDX,				\
+			(,),						\
+			node,						\
+			prop)						\
+}
+
+#define SENSING_DT_INFO(node)								\
+	static struct sensing_dt_info							\
+		_CONCAT(__sensing_dt_info_, Z_DEVICE_DT_DEV_ID(node)) = {		\
+			.dev = DEVICE_DT_GET(node),					\
+			.ord = DT_DEP_ORD(node),					\
+			.info.type = DT_PROP(node, sensor_type),			\
+			.info.name = DT_NODE_FULL_NAME(node),				\
+			.info.friendly_name = DT_PROP(node, friendly_name),		\
+			.info.minimal_interval = DT_PROP(node, minimal_interval),	\
+			.reporter_num = DT_PROP_LEN_OR(node, reporters, 0),		\
+			.reporters = PHANDLE_DEVICE_LIST(node, reporters),		\
+		};
+
+
+#define SENSING_DT_ITEM(node) &_CONCAT(__sensing_dt_info_, Z_DEVICE_DT_DEV_ID(node)),
```

Comment:
all above device tree suggestion is a good suggestion. Actually, we implemented sensor devicetree info just as you mentioned here. But we met one issue yperess has pointed out, yperess suggested not to define MAX REPROTER COUNT in sensing subsytstem KCONFIG. But if declare sensors_dt_list[] = {{.reporters[]}, {.reporters[]}... {.reporters[]}, ...}, compiler would error. So to remove MAX REPORTER COUNT macro defined in struct sensing_dt_info -> const struct device *reporters[]; we have to define static struct **sensing_dt_info** __sensing_dt_info_**node one by one, and then remove MAX REPORTER COUNT in reporters[], then compile would be success.

Reply:
I don't understand - I'm suggesting you do the following to declare the `struct sensor_dt_info`  directly into the array instead of declaring each sensor individually with a unique name, and then initializing an array to point to the static structures.

See my latest suggestions in sesnsor_mgmt.c.  I tried these changes out locally and the sample builds and runs successfully.